packages/common/src/utils/sources.ts

```diff
@@ -32,19 +25,16 @@ export const getSources = ({
   return sources.map((source: Source) => {
     const url = new URL(source.file);
 
-    const sourceUrl = url.href;
+    const isDRM = (url.searchParams.has('exp') && url.searchParams.has('sig')) || source.drm;
 
-    const isBCLManifest = isBCLManifestType(sourceUrl, baseUrl, mediaid, ['m3u8', 'mpd']);
-    const isVODManifest = isVODManifestType(sourceUrl, baseUrl, mediaid, ['m3u8', 'mpd']);
-    const isDRM = url.searchParams.has('exp') && url.searchParams.has('sig');
-
-    // Use SSAI URL for configs with server side ads, DRM is not supported
-    if (isVODManifest && hasServerAds && !isDRM) {
-      // Only HLS is supported now
-      url.href = `${baseUrl}/v2/sites/${siteId}/media/${mediaid}/ssai.m3u8`;
+    // Use SSAI URL for configs with server side ads, DRM is not supported yet
+    if (hasServerAds && !isDRM) {
+      url.href = `${baseUrl}/v2/sites/${siteId}/media/${mediaid}/ssai.${MANIFEST_TYPE.dash === source.type ? 'mpd' : 'm3u8'}`;
```

Comment:
@AntonLantukh we believe this now replaces external hosted URLs with the SSAI URL. This prevents playback because the SSAI URL yields an error.

Reply:
Hey @ChristiaanScheermeijer ! Could you elaborate on this one? I see we use a `APP_API_BASE_URL` env which should be `cdn`.

Reply:
I think it makes sense to check the `type` (mpd | m3u8) and make changes only for this case.